### PR TITLE
Add Grafana , Loki and fluentd for Logging

### LINKED ISSUE #12: Setup Grafana and Loki
Grafana and Loki should be installed to hardcase for shared logs.

### PR DESCRIPTION
**Pull request by butlerx**
Add grafana and loki service to hardcase

closes #12 

I've added a bare min to this pr but a follow up action to enable more parsers in fluentd to parse specific logs.

### CODE REVIEW

File: services/fluentd.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
+{ pkgs, config, lib, ... }:
+let
+  plugins = import ../packages/fluentd-plugins { inherit pkgs; };
+in {
+
+  services.fluentd = {
+    enable = true;
+    plugins = ["${plugins}/lib/ruby/gems/2.6.0/gems/fluent-plugin-grafana-loki-1.2.7/lib/fluent/plugin"];
```
Comment (butlerx) [inline services/fluentd.nix:8]: im not a big fan of this but im unsure is there a better way to get this

File: services/loki.nix [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+          index = {
+            prefix = "index_";
+            period = "168h";
+          };
+        }];
+      };
+
+      storage_config = {
+        boltdb = {
+          directory = "/tmp/loki/index";
```
Comment (m1cr0man) [inline services/loki.nix]: do we really want the storage for the service in /tmp?

File: services/loki.nix [hunk 3]
```diff
@@ -0,0 +1,79 @@
+{ config, lib, ... }:
+let
+  dataDir = "/var/lib/loki";
+in {
+
+  services.loki = {
+    inherit dataDir;
+    enable = true;
+    configuration = {
+      auth_enabledi = false;
```
Comment (m1cr0man) [inline services/loki.nix]: is the `i` a typo here?

**Review by m1cr0man — APPROVED** [review]
Looks good! 2 comments on things I don't understand, if you can reply to them I'll merge

File: services/loki.nix (hunk 3 above)
Comment (butlerx) [reply services/loki.nix]: Yes

File: services/loki.nix (hunk 2 above)
Comment (butlerx) [reply services/loki.nix]: no that was for testing lets get that out of there

**Pull request merged**